src/components/biggive-main-menu/biggive-main-menu.tsx

```diff
@@ -0,0 +1,131 @@
+import { Component, Element, Host, h, Method } from '@stencil/core';
+
+@Component({
+  tag: 'biggive-main-menu',
+  styleUrl: 'biggive-main-menu.scss',
+  shadow: true,
+})
+export class BiggiveMainMenu {
+  @Element() host: HTMLBiggiveHeaderElement;
+
+  openMobileMenu = () => {
+    const mobileMenu = this.host.shadowRoot.querySelector('.nav-links') as HTMLElement;
+    mobileMenu.style.left = '0';
+  };
+
+  closeMobileMenu = () => {
+    const mobileMenu = this.host.shadowRoot.querySelector('.nav-links') as HTMLElement;
+    mobileMenu.style.left = '-100%';
+  };
+
+  @Method()
+  async closeMobileMenuFromOutside() {
+    const mobileMenu = this.host.shadowRoot.querySelector('.nav-links') as HTMLElement;
+    mobileMenu.style.left = '-100%';
+  }
+
+  appendPrimaryNavigationLinks() {
+    // get the slotted 'nav-primary' node
+    const node = this.host.querySelector(`[slot="nav-primary"]`);
+
+    if (node !== null) {
+      // Slot the menu
+      this.host.shadowRoot.querySelector('#nav-primary').appendChild(node);
+    }
+  }
+
+  /**
+   * Slots the 'nav-secondary' slot into the top right corner of the blue nav bar.
+   * Also slots the 'nav-secondary' into the normal menu items when in mobile mode.
+   * This is because the blue bar disappears in mobile mode, so we need the items
+   * init to still be seen, hence they're auto-added to the menu.
+   */
+  appendNavSecondaryLinks() {
+    // get the slotted 'nav-secondary' node
+    const node = this.host.querySelector(`[slot="nav-secondary"]`);
+
+    if (node !== null) {
+      // add to blue bar
+      this.host.shadowRoot.querySelector('.nav-secondary').appendChild(node);
+
+      // we must make a deep clone of the node above, because each node is only
+      // injectable / slottable into one place, but we need to slot into two places.
+      const nodeClone = node.cloneNode(true) as HTMLElement;
+
+      // add to main menu links, but only viewable in mobile mode. The css class
+      // hides them in desktop!
+      nodeClone.querySelectorAll('li').forEach(child => {
+        child.classList.add('mobile-only');
+        this.host.shadowRoot.querySelector('.links').appendChild(child);
+      });
+    }
+  }
+
+  componentDidRender() {
+    const subMenuElements = this.host.querySelectorAll('.sub-menu') as NodeListOf<HTMLElement>;
+    subMenuElements.forEach(subMenuElement => {
```

Comment:
```suggestion
    this.host.querySelectorAll<HTMLElement>('.sub-menu-arrow').forEach(subMenuArrow => {
```

Reply:
Updated in [f8a7bbc](https://github.com/thebiggive/components/pull/99/commits/f8a7bbc1f42f976ff8ec567a1fc7568b47bee0a1) as discussed over Slack